hpolib/benchmarks/ml/svm_benchmark.py

```diff
@@ -0,0 +1,316 @@
+from sklearn.metrics import accuracy_score, make_scorer
+from sklearn.preprocessing import OneHotEncoder
+
+import hpolib.util.rng_helper as rng_helper
+from hpolib.abstract_benchmark import AbstractBenchmark
+from hpolib.util.openml_data_manager import OpenMLHoldoutDataManager
+
+__version__ = '0.0.1'
+
+
+class SupportVectorMachine(AbstractBenchmark):
+    """
+    Hyperparameter optimization task to optimize the regularization
+    parameter C and the kernel parameter gamma of a support vector machine.
+    Both hyperparameters are optimized on a log scale in [-10, 10].
+    The X_test data set is only used for a final offline evaluation of
+    a configuration. For that the validation and training data is
+    concatenated to form the whole training data set.
+    """
+
+    def __init__(self, task_id: Union[int, None] = None,
+                 rng: Union[np.random.RandomState, int, None] = None):
+        """
+        Parameters
+        ----------
+        task_id : int, None
+        rng : np.random.RandomState, int, None
+        """
+        super(SupportVectorMachine, self).__init__(rng=rng)
+
+        self.task_id = task_id
+        self.accuracy_scorer = make_scorer(accuracy_score)
+
+        self.X_train, self.y_train, self.X_valid, self.y_valid, self.X_test, self.y_test, variable_types = \
+            self.get_data()
+        self.categorical_data = np.array([var_type == 'categorical' for var_type in variable_types])
+
+        # Sort data (Categorical + numerical) so that categorical and continous are not mixed.
+        categorical_idx = np.argwhere(self.categorical_data)
+        continuous_idx = np.argwhere(~self.categorical_data)
+        sorting = np.concatenate([categorical_idx, continuous_idx]).squeeze()
+        self.categorical_data = self.categorical_data[sorting]
+        self.X_train = self.X_train[:, sorting]
+        self.X_valid = self.X_valid[:, sorting]
+        self.X_test = self.X_test[:, sorting]
+
+        nan_columns = np.all(np.isnan(self.X_train), axis=0)
+        self.categorical_data = self.categorical_data[~nan_columns]
+
+        mean_imputer = SimpleImputer(strategy='mean')
+        self.X_train = mean_imputer.fit_transform(self.X_train)
+        self.X_valid = mean_imputer.transform(self.X_valid)
+        self.X_test = mean_imputer.transform(self.X_test)
+
+        # Determine all possible values per categorical feature
+        complete_data = np.concatenate([self.X_train, self.X_valid, self.X_test], axis=0)
+        self.categories = [np.unique(complete_data[:, i])
+                           for i in range(self.X_train.shape[1]) if self.categorical_data[i]]
+
+        self.train_idx = self.rng.choice(a=np.arange(len(self.X_train)),
+                                         size=len(self.X_train),
+                                         replace=False)
+
+        # Use 10 time the number of classes as lower bound for the dataset fraction
+        n_classes = np.unique(self.y_train).shape[0]
```

Comment:
If this a heuristic you came up with or can we refer to some work for this?

Reply:
Aaron used it in his paper: 

[Fast Bayesian Optimization of Machine Learning Hyperparameters
on Large Datasets](https://arxiv.org/pdf/1605.07079.pdf)
> We implemented Hyperband following Li et al. (2017) using
the recommended setting for the parameter η = 3 that controls the intermediate subset sizes. For each experiment, we
adjusted the budget allocated to each Hyperband iteration to
allow the same minimum dataset size as for FABOLAS: 10
times the number of classes for the support vector machine
benchmarks and the maximum batch size for the neural
network benchmarks

Reply:
Sounds good, can you add a comment referencing that paper?